ambry-commons/src/main/java/com.github.ambry.commons/BlobId.java

```diff
@@ -463,6 +446,32 @@ public static BlobId craft(BlobId inputId, short targetVersion, short accountId,
         inputId.partitionId, inputId.uuid);
   }
 
+  /**
+   * Returns whether a given Blob id is a crafted id.
+   * @param idStr the blobId in string form.
+   * @return true if the id is a crafted id; false otherwise.
+   * @throws IOException if the input is not a valid Blob id.
+   */
+  public static boolean isCrafted(String idStr) throws IOException {
```

Comment:
Would it be better to just make the BlobIdPreamble class public so we don't have to introduce a method for any of these fields someone might want to check?

Reply:
I actually prefer keeping the concept of a preamble private within `BlobId`.